odk/src/main/java/org/motechproject/odk/parser/impl/XformParserKobo.java

```diff
@@ -0,0 +1,113 @@
+package org.motechproject.odk.parser.impl;
+
+import org.motechproject.odk.domain.FormDefinition;
+import org.motechproject.odk.domain.FormElement;
+import org.motechproject.odk.parser.XformParser;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import java.io.ByteArrayInputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class XformParserKobo extends XformParserODK implements XformParser {
+
+    private static final String GROUPS_PATH = "/h:html/h:body/xForms:group";
+    private static final String REF = "ref";
+    private static final String LABEL = "label";
+    private static final String REPEAT = "repeat";
+
+    public XformParserKobo() {
+        super();
+    }
+
+    @Override
+    public FormDefinition parse(String xForm, String configurationName) throws XPathExpressionException {
+        InputSource inputSource = new InputSource(new ByteArrayInputStream(xForm.getBytes()));
+        Node root = getRoot(inputSource);
+        FormDefinition formDefinition = parseXForm(configurationName, root);
+        formDefinition.setXform(xForm);
+        findGroupLabels(formDefinition, root);
+        return formDefinition;
+
+    }
+
```

Comment:
can we make the breaks between the methods consistent? sometimes you use two lines, sometimes one